pkg/controller/configuration/template_merger.go

```diff
@@ -94,21 +94,21 @@ func (c *configPatcher) Merge(baseData map[string]string, updatedData map[string
 		return baseData, nil
 	}
 
-	r := make(map[string]string)
+	mergedData := copyMap(baseData)
 	params := core.GenerateVisualizedParamsList(configPatch, formatter, nil)
 	for key, patch := range splitParameters(params) {
```

Comment:
Seems this logic doesn't handle configPatch.DeleteConfig in params properly.

Reply:
I see, the configPatch object returned by core.TransformConfigPatchFromData should only contain UpdateConfig